File: bupt-hotel-frontend/src/types/Room.ts
// src/types/Room.ts

export type Wind = 'slow' | 'mid' | 'high' | 'stop';

/**
 * 后端返回的房间监控详情
 */
export interface RoomDetail {
  /** 房间号 */
  id: number;
  /** 客人姓名，空闲时为 null */
  guest: string | null;
  /** 当前温度 */
  currentTemp: number;
  /** 目标温度 */
  targetTemp: number;
  /** 当前风速 */
  currentSpeed: Wind;
  /** 目标风速 */
  targetSpeed: Wind;
  /** 耗电量（度） */
  electricalUsage: number;
  /** 累计费用（元） */
  cost: number;
  /** 空调开关状态，true=开，false=关 */
  status: boolean;
  /** 房间占用状态，1=占用，0=空闲 */
  roomStatus: number;
}

File: bupt-hotel-backend/src/main/java/com/e303/hotel/bean/Room.java
package com.e303.hotel.bean;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.e303.hotel.bean.enums.Speed;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@TableName("room")
public class Room {
    @TableId(value = "room_id", type = IdType.AUTO) // 👈 这行很关键！
    //@JsonProperty("room_id")
    private Integer roomId;
    private String password;
    private Integer roomStatus; // 0: 空闲, 1: 不空闲
    private Integer status;     // 0: 关闭, 1: 开启
    private float currentTemp;
    private Speed currentSpeed; //"high" "mid" "slow" "stop"
    private float targetTemp;
    private Speed targetSpeed;  // "high" "mid" "slow" "stop"
    private float electricalUsage;
    private float fee;
    //private float initTemp = 25.0f;

    public float getInitialTemp(){
        return 25.0f;
    }


}
